### ISSUE TITLE
hightlight wrong when use global namespace symbol

### ISSUE THREAD

**Issue by changnet** [atom/language-c#80]
![2015-07-04 19 45 49](https://cloud.githubusercontent.com/assets/9293777/8507779/4e951492-2285-11e5-8abc-6fadd30a4832.png)

**Comment by 50Wliu** [comment]
Are you sure the problem isn't actually happening on this line?
```c
::printf("Error: MongoDB Invalid Query %s!\n", query ? query->toString().c_str() : "{}");
```
It looks like the brackets at the end are interfering with string highlighting.

**Comment by changnet** [comment]
it do happening on this line,no matter global namespace symbol(::) exist or not
```C++
::printf("Error: MongoDB Invalid Query %s!\n", query ? query->toString().c_str() : "{}");
```
but the code below
```C++
{
    ::printf("Error: MongoDB Invalid Query %s!\n", query ? query->toString().c_str() : "{}");
}
```
it hightlighting right without global namespace symbol
![20150706113944](https://cloud.githubusercontent.com/assets/9293777/8515383/1cde94be-23d4-11e5-9ffc-a72c9fc708d9.jpg)

**Comment by 50Wliu** [comment]
Thanks for the extra info, @changnet!  I can reproduce this now.

**Comment by dixx** [comment]
@50Wliu It seems to me that the ternary operator is not checked correctly in all cases (language-c), so that method calls on pointers within `?` and `:` cause trouble.
This will work fine: `x = a ? b : c;`, while this `x = a ? b->f() : c;` will mess up the highlighting of the code behind it to a certain degree. This `x = a ? b->f( d->f() ) : c;` will mess it up even further. I hope that helps to catch this bug!

**Comment by dixx** [comment]
And this issue seems to be related to #178.

**Comment by 50Wliu** [comment]
This has been fixed, most likely by #179.
